clang/lib/Driver/Driver.cpp

```diff
@@ -5524,9 +5524,11 @@ InputInfo Driver::BuildJobsForActionNoCache(
   else {
     std::string OffloadingPrefix;
     // When generating binaries with -fsycl-link-target or -fsycl-link, the
-    // output file prefix is the triple arch only.
-    if (Args.getLastArg(options::OPT_fsycl_link_targets_EQ) ||
-        Args.hasArg(options::OPT_fsycl_link_EQ)) {
+    // output file prefix is the triple arch only.  Do not add the arch when
+    // compiling for host.
```

Comment:
```suggestion
    // compiling for non-host.
```
(if I understand the changes correctly)

Reply:
Here we are only adding the arch for non-host (correct) but the wording is to 'not add' for host.

Reply:
Ah, my bad, it finally came together :D